Improvements to logo-blue SVG-structure

GammaSQ opened the pull request:
Fixes https://github.com/QubesOS/qubes-issues/issues/9678

The qubes-logo is now consisting of only three paths (one for each color) which are perfectly aligned at their edges. The Q-"comma" is now part of the lower right path, not a rectangle.

Mainly intended for tasks other than design, such as milling, cutting, and so forth. Changes are unlikely to affect visual representations.

marmarek commented:
There looks to be a 1px gap in bottom part

GammaSQ commented:
Thank you for reminding me why I'm a programmer and not a designer. I see what you mean.

According to inkscape, there is no gap. both shapes end at the exact same coordinates, with a straight line between them. However, exporting a .png reveals there is actually a blurry gap between *all* shapes. The blue background hexagon was probably a workaround for exactly this problem. (increasing the size of the "document" helped a little, disabling anti-aliasing removed the gap, but then the edges look fuzzy)

Anyway, this is as far as I can help. There is probably some way to tell SVG two shapes should be touching, but I don't know it. I believe this was the reason why the mid-blue-part used to be a full hexagon. Breaking the logo into distinct shapes however is the whole point, so I hope there is a different setting/property to achieve original beauty with separate shapes.

GammaSQ commented:
setting "blend mode" to "color-burn" seems to work.

marmarta reviewed (approved):
looks good to me, I checked it in a couple of programs and it looked well

marmarta commented:
Hmm, no, wait, at some sizes I get a dark line between shapes ...

marmarta commented:
E.g. inkscape - export to 256 x 256 pixels

GammaSQ commented:
All shapes have a border-width of 1px now. Looks good in all exports I tried.

Downsides: The visual border between two shapes is shifted by 1px. There is a tiny visible kink at the curved edges. It's practically invisible unless extremely zoomed in. For very high resolution exports, I recommend removing the border again.

However, the export-artifacts I observed did not include dark lines. I hope the 1px border fixes that as well, but please confirm!

marmarta commented:
Ok, I think it works - could you squash the commits into one commit?

GammaSQ committed:
Added border to all shapes in order to blend bordering colors together correctly

The qubes-logo is now consisting of only three paths
(one for each color) which are perfectly aligned at their edges.
The Q-"comma" is now part of the lower right path, not a rectangle.

Mainly intended for tasks other than design, such as milling,
cutting, and so forth. Changes are unlikely to affect visual
representations.

GammaSQ commented:
I hope this worked ...

marmarta commented:
signature is missing: https://www.qubes-os.org/doc/code-signing/
